### ISSUE TITLE
VoiceOver does not identify Calcite Dropdown as a dropdown menu

### ISSUE THREAD

**Issue by Anastasiia-Boleiko** [Esri/calcite-design-system#12520]
**monday.com sync:** #18005841859

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

When utilizing VoiceOver (uncertain about other screen readers), the announcement when navigating to the Calcite Dropdown does not indicate anything about the dropdown itself. Users need to be informed that it is not just a button but contains a menu by including appropriate roles, ARIA properties, or other elements to ensure that assistive technologies identify the component accurately. This issue persists even though I have identified the necessary ARIA properties in the DOM structure of the Calcite Dropdown component. However, I am uncertain whether they are correctly specified on the DOM element.

<img width="2280" height="808" alt="Image" src="https://github.com/user-attachments/assets/dd96ea2f-d028-4a3c-997c-acd38dbbb166" />

### Actual Behavior

VoiceOver does not provide any information about the dropdown menu: 
<img width="800" height="323" alt="Image" src="https://github.com/user-attachments/assets/e380ba2d-b92f-4c81-871e-3c9860e4eca1" />

<img width="1866" height="754" alt="Image" src="https://github.com/user-attachments/assets/5cd3b233-872b-4f1c-af97-a3b12f828b0b" />

### Expected Behavior

VoiceOver indicates that the button functions as a dropdown menu and also states its current state (collapsed/expanded), and so on.

### Reproduction Sample

https://codepen.io/anastasiia-zrh/pen/JoYYdNz?editors=100

### Reproduction Steps

1. Open any example of the Calcite Dropdown.
2. Enable VoiceOver.
3. Navigate to the dropdown (ex. using keyboard focus)
4. Check what VoiceOver announces when reaching the dropdown.

### Reproduction Version

Calcite 3.2.1

### Working W3C Example/Tutorial

https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/

### Relevant Info

Example from w3.org:

<img width="480" height="212" alt="Image" src="https://github.com/user-attachments/assets/fb013414-c1d0-4dd1-8a34-38db033aec9d" />
<img width="480" height="212" alt="Image" src="https://github.com/user-attachments/assets/25b3aca2-096f-416b-8611-353b97b49f03" />
<img width="480" height="212" alt="Image" src="https://github.com/user-attachments/assets/7df806d5-9e95-4195-a0cf-b59c3662ee5f" />

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Urban

**Comment by geospatialem** [comment]
### Research findings

While the context of the element slotted in the `dropdown`'s  is not providing the context the component is a dropdown, this is expected behavior in alignment with other implementations across the web, as the slotted element should be communicated to assistive technologies. For instance: [Orange's accessible example](https://a11y-guidelines.orange.com/en/articles/dropdown-menu/).

#### Key takeaways
- The `trigger` slot is planned for deprecation and removal, timing tbd based on [the dropdown type and slot deprecation](https://github.com/Esri/calcite-design-system/issues/14173)
    - Example implementation/workaround: https://codepen.io/geospatialem/pen/NPbRWpq from changes in https://github.com/Esri/calcite-design-system/issues/12426 for `5.1`
- The original request is out of scope, but the component used as the component's `referenceElement` (previously slotted in `trigger`) should leverage an `aria` property similar to action

Closing the above as unplanned in favor of https://github.com/Esri/calcite-design-system/issues/13912 and specifically https://github.com/Esri/calcite-design-system/issues/14404. 

### Summary
While the team works through prioritization the above can be achieved as [a workaround when a `calcite-button` is used as either the `dropdown`'s `referenceElement` or slotted `trigger` element](https://codepen.io/geospatialem/pen/NPbRWpq) (_note: soon to be deprecated_).

**Comment by Anastasiia-Boleiko** [comment]
@geospatialem Thank you for the research and for creating another ticket for aria-expanded attribute.
